ports/nuraft/fix-build-compatibility-issues.patch

```
@@ -0,0 +1,80 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 0e5ac72..59a0032 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -170,7 +170,12 @@ else()
+     # === Other shared libraries ===
+     if(NOT WIN32)
+         set(LIBDL dl)
+-        set(LIBZ z)
++        find_package(ZLIB QUIET)
++        if(ZLIB_FOUND)
++            set(LIBZ ZLIB::ZLIB)
++        else()
++            set(LIBZ z)
++        endif()
```

```suggestion
+        find_package(ZLIB REQUIRED)
+        set(LIBZ ZLIB::ZLIB)
```

It's a dependency so it will be found and if not it's a serious bug that should fail early.

---

This was patched upstream. Tracking link: [eBay/NuRaft#627](https://github.com/eBay/NuRaft/pull/627).